src/pages/patientView/trialMatch/TrialMatchTableUtils.ts

```diff
@@ -140,15 +158,22 @@ export function calculateTrialPriority(armMatches: IArmMatch[]): number {
 export function getMatchPriority(clinicalGroupMatch: IClinicalGroupMatch): number {
     // In trial match tab, positive matches should always display before negative matches(notMatches).
     // The highest and default priority is 0. The priority the higher, the display order the lower.
-    if (clinicalGroupMatch.notMatches.length > 0) {
-        if (clinicalGroupMatch.matches.length === 0) {
+    const matchesLength = getMatchesLength(clinicalGroupMatch.matches);
+    const notMatchesLength = getMatchesLength(clinicalGroupMatch.notMatches);
+    if (notMatchesLength > 0) {
+        if ( matchesLength === 0) {
             return 2; // A trial only has negative matches.
         }
         return 1; // A trial has both positive matches and negative matches.
     }
     return 0; // A trial only has positive matches.
 }
 
+export function getMatchesLength(genomicMatchType: IGenomicMatchType): number {
+    return _.sum([genomicMatchType.MUTATION.length, genomicMatchType.CNA.length,
```

Comment:
you don't have to make this change, but why not just take sum of any array found in the collection?  you could do this with _.chain(genomicMatchType).map().sumBy(a=>a.length).
again, you won't have to worry in future about chaning this code.